frontend/src/features/Products/components/Products.tsx

```diff
@@ -1,18 +1,15 @@
 import React from "react";
 
-import { connector, PropsFromRedux } from "../connector";
 import { List } from "components/List";
+import { useProducts } from "../hooks/useProducts";
 
-type Props = PropsFromRedux & {
+type Props = {
   myProp: string;
 };
 const Products = (props: React.PropsWithChildren<Props>) => {
-  const { onGetProducts } = props;
-  React.useEffect(() => {
-    onGetProducts();
-  }, [onGetProducts]);
+  const { products } = useProducts();
```

Comment:
Responsiblities have slightly changed here. Component lifecycle events are now handled by hook. The argument in doing this is that presentational components only receive the absolute minimum data and callbacks that they need for rendering. Logic to get the data is in the hook. Advantage in this is that it makes presentational component simpler and the functionality easier to test since you only need to test the hook. Disadvantage is that `useProducts` hook might be less reusable since it asumes that we want to fetch all products on mount.